site/docs/v1/tech/admin-guide/configuration-management/terraform.adoc

```diff
@@ -0,0 +1,502 @@
+====
+
+We suggest to leave the import block in your configuration, a record of the resource's origin. The import block records that Terraform imported the resource and did not create it.
+
+=== Terraform Data Source Instead Of Import
+
+Instead of importing a resource you can use an existing FusionAuth configuration as a `Data Source`, the list of all supported data sources you can find in the link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs[FusionAuth Terraform Provider Documentation]
+
+In the scenario where you choose to manage the Default tenant and FusionAuth application outside Terraform, but still add applications in to the default Tenant via Terraform, you can reference it in the configuration as a data source.
+
+* link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/data-sources/tenant[Terraform Data Source: Application]
+* link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/data-sources/application[Terraform Data Source: Application]
+
+[source]
+----
+data "fusionauth_tenant" "Default" {
+  name = "Default"
+}
+
+data "fusionauth_application" "FusionAuth" {
+  name = "FusionAuth"
+}
+----
+
+[NOTE.note]
+====
+If you manage the Default tenant and FusionAuth application outside of Terraform but want specific actions (scripts, API calls, etc.) still integrated and triggered by your Terraform configuration, you could make the use of link:https://developer.hashicorp.com/terraform/language/resources/provisioners/syntax[Provisioners].
+
+Terraform includes the concept of provisioners as a measure of pragmatism and last resort (Since provisioners are non-declarative and potentially unpredictable), knowing that there are always certain behaviors that cannot be directly represented in Terraform's declarative model.
+====
+//TODO Provisioner https://developer.hashicorp.com/terraform/language/resources/provisioners/syntax
+
+== Terraform Create
+
+To create a new resource you can go through the list of resources available to you in the link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs[FusionAuth Terraform Provider Documentation] and pick the resource you are interested in. Each resource contains information about required and optional arguments.
+
+In this example we create an application called forum in the Default tenant with related roles (admin and user), users (forum-user1 and forum-admin1) and according registrations:
+
+* link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/resources/application[Terraform Resource: Application]
+* link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/resources/application_role[Terraform Resource: Application Role]
+* link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/resources/user[Terraform Resource: User]
```

Comment:
Its not typical for production environments for sure, but for Test it could be valid. We for example use it to stage the test environment with some default users and additional stuff like fusionauth_registration with prefilled custom data.

With regards to the example I wanted to have something simple which then gets modified very frequently (in the case of user even the first login) to have a easy example on how to use ignore_changes. 

But I can figure out something else instead, as mentioned later the FusionAuth Application data field works similar, it is something you create with terraform but should ignore if you make use of it with your end user application.

Reply:
Makes sense. Let's use application and application.data as something to ignore, please.